src/utils/propUtils.js

```diff
@@ -10,36 +10,83 @@ import Image from '../elements/Image/Image'
 // ===============================================================
 export const customPropTypes = {
   /**
-   * Ensures children are of a set of types.
-   * Similar to `oneOfType` built-in validator
-   * @param {String[]} allowedTypes Collection of allowed Stardust component types
+   * Ensures children are of a set of types. Matches are made against the component _meta.name property.
+   * @param {String[]} allowedTypes Collection of allowed component types.
    */
   ofComponentTypes: (allowedTypes) => {
     return (props, propName, componentName) => {
-      const { children } = props
-      const disallowed = Children.map(children, child => {
+      if (propName !== 'children') {
+        throw new Error(`ofComponentTypes can only be used on the \`children\` prop, not ${propName}`)
+      }
+      if (!_.isArray(allowedTypes)) {
+        throw new Error([
+          'Invalid argument supplied to ofComponentTypes, expected an instance of array.'
+            ` See ${componentName} prop \`${propName}\`.`,
+        ].join(''))
+      }
```

Comment:
Added validation to all custom prop types arguments.  Some of these were being improperly used, so there was no validation happening at all.